### DISCUSSION TITLE
Duplicate Posts After Using Trim Reblogs

### DISCUSSION THREAD

**Discussion by stasisity** [AprilSylph/XKit-Rewritten#611] [Help Desk]
Hello!
Having some complications here. When I trim a drafted post, it will trim the post but duplicate my response. Any idea what might be causing this?

**Comment by AprilSylph** [comment]
How are you drafting the posts initially? (This lines up with what happens if you try to trim a legacy post, but I thought I'd put in ample protection against that.)

**Reply by stasisity** [reply]
It's a little bit more complicated then that, but yes! The thread was started in Legacy. The thread has gone like this so far:
Legacy > Mobile > Legacy (w/ New-Kit Editable Reblogs) > Mobile > Legacy (w/ xkit rewritten) > mobile > legacy (w/ xkit rewritten) > mobile > beta (w/ xkit rewritten)

However, if I try to replicate the thread order, the problem occurs as early as 'Legacy (w/ xkit rewritten)'. So I may have missed something, or I'd probably ignored the duplication issue before and simply deleted the duplicated response in legacy.
I also made note that Once the switch was made to xkit-rewritten, the hyperlink for my url was removed and the icon set to default.
![image](https://user-images.githubusercontent.com/105946383/169661429-7812be61-1f7c-4e05-90f1-8dfd05e1ef60.png)

I'll give you some more details/specs on my browser to better understand what might be bypassing your protections.
It does _not_ appear to be linked to any addons that I have enabled, as the problem can still be replicated without them. But I'll include a list anyway.

Windows 10 Home
Firefox 100.0.2 (64-bit)
**Active Addons**
- AdNauseam
- Decentraleyes
- uBlock Origin
- Keeper® Password Manager & Digital Vault
- Firefox Multi-Account Containers
- RPThreadTracker QuickAdd
- Facebook Container
- Don't track me Google
- Privacy Badger
- Grammar & Spell Checker — LanguageTool
- New-Xkit

Tumblr Dashboard Settings
![image](https://user-images.githubusercontent.com/105946383/169661355-a51f9b71-a059-45ea-a93d-da04d0b217c7.png)


Anyways, I absolutely understand now that I _definitely_ need to start threads and continue them in Beta from now on if I intend to use xkit-rewritten. I'm just hoping this troubleshooting session can help you in developing it further!

**Reply by AprilSylph** [reply]
Only the root post is checked for being legacy, so if it's allowing you to trim where the root post is legacy, that's a bug. I actually tested what happens if you reblog an NPF post with legacy methods, and it should work fine.

The Editable Reblogs incompatibility isn't to do with legacy reblogging so much as it's about how it removes the entire trail, so Trim Reblogs has nothing to trim.
